c10/util/Backtrace.cpp

```diff
@@ -87,6 +95,47 @@ void dump_stack(
 #if SUPPORTS_BACKTRACE
 namespace {
 
+#if !defined(C10_ANDROID) && !defined(__APPLE__)
+
+// converts a function's address in memory to its VMA address in the executable
+// file. VMA is what addr2line expects
+size_t ConvertToVMA(size_t addr) {
+  Dl_info info;
+  link_map* link_map;
+  dladdr1((void*)addr, &info, (void**)&link_map, RTLD_DL_LINKMAP);
+  return addr - link_map->l_addr;
+}
+
+std::string exec(const char* cmd) {
+  std::array<char, 128> buffer;
+  std::string result;
+  std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd, "r"), pclose);
+  if (!pipe) {
+    throw std::runtime_error("popen() failed!");
+  }
+  while (fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr) {
+    result += buffer.data();
+  }
+  return result;
+}
+
+std::string rstrip(const std::string& s) {
+  const std::string WHITESPACE = " \n\r\t\f\v";
+  size_t end = s.find_last_not_of(WHITESPACE);
+  return (end == std::string::npos) ? "" : s.substr(0, end + 1);
+}
+
+bool use_addr2line() {
+  static bool _use_addr2line = []() {
+    return c10::utils::check_env("TORCH_SHOW_CPP_STACKTRACES_WITH_LINENO") ==
+        true &&
+        !system("which addr2line > /dev/null 2>&1");
```

Comment:
The addr2line availability test is probably not necessary, because you just won't pass `TORCH_SHOW_CPP_STACKTRACES_WITH_LINENO` if addr2line doesn' twork

Reply:
fixed.